### DISCUSSION TITLE
Switch to GPU?

### DISCUSSION THREAD

**Discussion by sharma2007** [ultralytics/ultralytics#826] [Q&A] [answered]
I'm training a model using the Python SDK. Unfortunately, it always uses my CPU... how do I make it switch to my GPU? (I have a RTX 3050 Ti).
Thanks!

**Comment by AyushExel** [comment]
@sharma2007 this might be a bug as we automatically detect cuda:0 device if its available. Try passing `device="0"` with model.train() and let me know if you still see this problem.
But first you should confirm what this returns:
```
imort torch

print(torch.cuda.is_available())
```
This should return true if your cuda installation is correct

**Reply by sharma2007** [reply]
Yeah, so doing this returns false. I have cuda and cudnnn both on my device with all the required things in the PATH... any ideas where am I going wrong?

**Reply by sharma2007** [reply]
When checking the version of cuda through cmd using 'nvcc --version', it shows I have cuda 11.8:
![image](https://user-images.githubusercontent.com/92726699/217006804-d0ec806b-045e-48a8-a795-17e0c73f2169.png)

**Reply by glenn-jocher** [reply, marked as answer]
@sharma2007 if `torch.cuda.is_available()` returns `False`, it typically indicates a mismatch between the installed PyTorch version and the CUDA version, or an issue with the CUDA installation itself. Since you've confirmed that CUDA 11.8 is installed, please ensure that you have the correct version of PyTorch that is compatible with CUDA 11.8.

You can install the appropriate PyTorch version from the official PyTorch website by selecting the correct specifications for your system and CUDA version. Once you've installed the compatible version, run the `torch.cuda.is_available()` check again.

If the issue persists, you might need to reinstall CUDA and ensure that all environment variables are correctly set. Double-check the NVIDIA driver version as well, as it needs to be compatible with the CUDA version you're using.
